image_segmentation/pytorch/capture_profile.py

```diff
@@ -0,0 +1,72 @@
+import argparse
+
+import torch_xla.debug.profiler as xp
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Performs an on-demand profiling session on provided profiler servers."
+    )
+
+    parser.add_argument(
+        "--service_addr",
+        dest="service_addr",
+        type=str,
+        required=True,
+        help='comma delimited string of addresses of the profiling servers to profile. ex. "10.0.0.2:8466" or "localhost:9012".',
+    )
+    parser.add_argument(
+        "--logdir",
+        dest="logdir",
+        type=str,
+        required=True,
+        help='the path to write profiling output to. Both the profiler client and server must have access. ex. "gs://bucket/file/path".',
+    )
+    parser.add_argument(
+        "--duration_ms",
+        dest="duration_ms",
+        type=int,
+        default=30000,
+        help="duration in milliseconds for tracing the server.",
+    )
+    parser.add_argument(
+        "--interactive",
+        dest="interactive",
+        type=str,
+        choices=[None, "once", "loop"],
+        default=None,
+        help=(
+            "run in interactive mode.\n"
+            'If set to "once", the profiler client asks for user confirmation before starting profiling.\n'
+            'If set to "loop", the profiler client repeatedly runs profiling, asking for user confirmation on each run.\n'
+            "Defaults to None, which disables interactive mode."
+        ),
+    )
+
+    flags = parser.parse_args()
+
+    def trace():
+        xp.trace(
+            service_addr=flags.service_addr,
+            logdir=flags.logdir,
+            duration_ms=flags.duration_ms,
+        )
+        print(f"Saved profiling output to {flags.logdir}")
+
+    def request_user_confirmation():
+        input("Press enter to start profiling:")
+
+    # Run performance profiling
+    if flags.interactive == "once":
+        request_user_confirmation()
+        trace()
+    elif flags.interactive == "loop":
+        while True:
```

Comment:
Nice feature. We can make it even better.

I suggest 

+ we offer a `for` loop mechanism (i.e. another `elif`) to the user to capture `N` profiles, every `M` seconds, for `T` seconds.

Reply:
Second suggestion, can we add a flag that offers a profile `start_time` capability to the user?

+ This is useful if we want the `once` profile to be captures after `A` seconds or `B` epochs. Wdyt?

+ This is equally useful for the `for` loop mechanism.

Reply:
Thanks for the suggestions @miladm 

I have added a single flag `--automatic` that achieves the `for` loop mechanism you are suggesting.
- `--automatic` requires 2 arguments. The 1st argument specifies the number of profiles to capture. The 2nd argument specifies the time gap (in seconds) between the profiles. Example: `--automatic 100 60` captures 100 profiles every 60 seconds.
- The previous `--duration_ms` argument already takes care of your `T` seconds suggestion.
- `--automatic` defaults to `None`, which disables automatic mode.

I have also added an optional `--start_time` flag that achieves your second suggestion.

I have tested `capture_profile.py` in `--automatic` mode and confirmed it worked as expected.